### PR TITLE
Support for refresh tokens

### PR DESCRIPTION
**Pull request by colq2**
According to #12 and #106 I updated this package to support refresh token.

What did changed:
First of all I added some options. The important one is the `refreshType`. There are three types:
1. (httponly): refresh token is set via HttpOnly Cookie which is the safest method
2. (storage): refresh token is safe in the storage
3. (null): refresh token is not use

As a result there is now a response inteceptor which is used to refresh the token automatically. 

There are setters and getters for the refresh token and the expiration.

Also I outsourced some things to utils to make the code more readable.

### CODE REVIEW

File: src/authenticate.js [hunk 1]
```diff
@@ -179,18 +310,58 @@ export default class VueAuthenticate {
...
+    requestOptions = makeRequestOptions(requestOptions, this.options, 'refreshUrl', data)
+    return this.$http(requestOptions)
+      .then((response) => {
+        this.setToken(response)
+        this.setRefreshToken(response)
+        return response
+      })
+      .catch((error) => {
+        this.clearStorage()
+        return error;
```
Comment (colq2) [inline src/authenticate.js]: This should be `return Promise.reject(error)`

File: src/authenticate.js [hunk 2]
```diff
@@ -179,18 +310,58 @@ export default class VueAuthenticate {
...
+
+    if (this.options.refreshType === 'storage')
+      data.refresh_token = this.getRefreshToken()
+
+    requestOptions = makeRequestOptions(requestOptions, this.options, 'refreshUrl', data)
+    return this.$http(requestOptions)
+      .then((response) => {
+        this.setToken(response)
+        this.setRefreshToken(response)
+        return response
```
Comment (colq2) [inline src/authenticate.js]: This should be `return Promise.resolve(response)`

**Comment by dgrubelic** [comment]
Thanks for your PR, i'll review it and write you my comments.

**Comment by matrunchyk** [comment]
Can you review it @dgrubelic?

**Comment by Andrei-Pozolotin** [comment]
+1

**Comment by youleaf** [comment]
+1

**Comment by pnuckowski** [comment]
Any update in this PR ? :)

**Pull request closed without merging**